main.c

```diff
@@ -0,0 +1,91 @@
+#include <string.h>
+#include "main.h"
+#include "_cgo_export.h"
+
+result *save_to_png(size_t max_store, int page_number, int width, float scale, const unsigned char *payload, size_t payload_length) {
+	if (max_store == 0) {
+		max_store = FZ_STORE_DEFAULT;
+	}
+
+	fz_context *ctx = fz_new_context(NULL, NULL, max_store);
+	if (!ctx) {
+		errorHandler("fail to create mupdf context");
+		return NULL;
+	}
+
+	result *r = malloc(sizeof(result));
+	r->context = ctx;
+	r->buffer = NULL;
+
+	int exit = 0;
+	fz_stream *stream = NULL;
+	fz_document *doc = NULL;
+	fz_page *page = NULL;
+	fz_display_list *list = NULL;
+	fz_device *device = NULL;
+	fz_pixmap *pixmap = NULL;
+	fz_device *draw_device = NULL;
+	fz_buffer *buffer = NULL;
+	fz_try(ctx) {
+		fz_register_document_handlers(ctx);
+		stream = fz_open_memory(ctx, payload, payload_length);
+		doc = fz_open_document_with_stream(ctx, "document.pdf", stream);
+		page = fz_load_page(ctx, doc, page_number);
+
+		float scale_factor = 1;
+		fz_rect bounds = fz_bound_page(ctx, page);
+		if (width != 0) {
+			scale_factor = width / bounds.x1;
+		} else if (scale != 0) {
+			scale_factor = scale;
```

Comment:
The usage of width & scale parameters in a way that can override each here is confusing. Why have them both and with an override mechanism ? Can't you always pass the width parameter only and decide scale here internally?

Reply:
This part I copied from the current lazypdf version: https://github.com/Nitro/lazypdf/blob/master/faster_raster.go#L359-L371

At the moment I don't want to change the current behaviour as it's working as expected.